src/src/PluginStructs/P159_data_struct.cpp

```diff
@@ -335,23 +374,39 @@ bool P159_data_struct::plugin_get_config_value(struct EventStruct *event,
 
   if (isValid()) {
     const String  cmd   = parseString(string, 1);
-    const int16_t count = logAll && !_engineeringMode ? P159_NR_OUTPUT_OPTIONS : P159_NR_ENGINEERING_OUTPUT_OPTIONS;
+    const int16_t count = P159_NR_MAX_OUTPUT_OPTIONS;
 
     for (int16_t option = 0; option < count; ++option) {
       int16_t idx  = option;
       int16_t gate = -1;
 
+      if ((option >= P159_OUTPUT_STATIC_SENSOR_ENERGY_GATE0) && (option <= P159_OUTPUT_STATIC_SENSOR_ENERGY_GATE8)) {
+        idx  = P159_OUTPUT_STATIC_SENSOR_GATE_index;
+        gate = option - P159_OUTPUT_STATIC_SENSOR_ENERGY_GATE0;
+      } else
+      if ((option >= P159_OUTPUT_MOVING_SENSOR_ENERGY_GATE0) && (option <= P159_OUTPUT_MOVING_SENSOR_ENERGY_GATE8)) {
+        idx  = P159_OUTPUT_MOVING_SENSOR_GATE_index;
+        gate = option - P159_OUTPUT_MOVING_SENSOR_ENERGY_GATE0;
+      } else
       if ((option >= P159_OUTPUT_STATIC_DISTANCE_ENERGY_GATE0) && (option <= P159_OUTPUT_STATIC_DISTANCE_ENERGY_GATE8)) {
-        idx  = P159_OUTPUT_STATIC_DISTANCE_GATE_index;
-        gate = option - P159_OUTPUT_STATIC_DISTANCE_ENERGY_GATE0;
+        if (_engineeringMode) {
+          idx  = P159_OUTPUT_STATIC_DISTANCE_GATE_index;
+          gate = option - P159_OUTPUT_STATIC_DISTANCE_ENERGY_GATE0;
+        } else {
+          idx = -1;
+        }
       } else
       if ((option >= P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE0) && (option <= P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE8)) {
-        idx  = P159_OUTPUT_MOVING_DISTANCE_GATE_index;
-        gate = option - P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE0;
+        if (_engineeringMode) {
+          idx  = P159_OUTPUT_MOVING_DISTANCE_GATE_index;
+          gate = option - P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE0;
+        } else {
+          idx = -1;
+        }
       }
 
-      if (logAll ||
-          cmd.equalsIgnoreCase(concat(Plugin_159_valuename(idx, false), (gate > -1) ? String(gate) : F("")))) {
+      if ((idx > -1) && (logAll ||
+                         cmd.equalsIgnoreCase(concat(Plugin_159_valuename(idx, false), (gate > -1) ? String(gate) : F(""))))) {
```

Comment:
Only use the `?` ... `:` operator when both have the same return type.
`String` and `F()` are not the same type.